src/methods/SANA.cpp

```diff
@@ -1106,9 +1106,37 @@ void SANA::performChange(int type) {
             item.second += localScoreSumIncChangeOp(localSimMatrixMap[item.first], source, oldTarget, newTarget);
     }
 
+
+
     double newCurrentScore = 0;
     bool makeChange = scoreComparison(newAligEdges, newInducedEdges, newTCSum, newLocalScoreSum, newWecSum, newNcSum, newCurrentScore, newEwecSum, newSquaredAligEdges, newExposedEdgesNumer);
+
+    if(wasBadMove && !makeChange)
```

Comment:
I think the only thing we want to find out is which hole is the better place to be; set a variable equal to either source or newTarget; and then have the code inside the #if CORES only once, using that variable.